src/acquisition/covid_hosp/state_daily/update.py

```diff
@@ -21,7 +23,51 @@ def run(network=Network):
       Whether a new dataset was acquired.
     """
 
-    return Utils.update_dataset(Database, network)
+    #return Utils.update_dataset(Database, network)
+
+    # can't use Utils here because daily files are posted in batches and we want
+    # all files in the batch. These have to be scraped from the Revisions page,
+    # since they aren't surfaced in metadata. :(
+    # then we merge them into an issue based on their publish date.
+
+    # get dataset details from metadata
+    metadata = network.fetch_metadata()
+    url, revision = Utils.extract_resource_details(metadata)
+    issue = Utils.get_issue_from_revision(revision)
+    print(f'issue: {issue}')
+    print(f'revision: {revision}')
+
+    # connect to the database
+    with Database.connect() as db:
+
+      # bail if the dataset has already been acquired
+      if db.contains_revision(revision):
+        print('already have this revision, nothing to do')
+        return False
+
+      max_issue = db.get_max_issue()
+      
+      # add metadata to the database
+      metadata_json = json.dumps(metadata)
+      db.insert_metadata(issue, revision, metadata_json)
+
+      urls = network.fetch_revisions(max_issue) + [url]
+      print(f'acquiring {len(urls)} daily updates')
+      n = 0
+      dataset = None
+      for url in urls:
```

Comment:
how confident are we that each `new` will have the same index/row order so that `update()` doesn't overwrite the wrong thing?

also as an alternative implementation, could use functools.reduce and do
```
dfs = [network.fetch_dataset(url) for url in urls]
dataset = reduce(lambda x, y: x.update(y),  dfs)
```

Reply:
it's very likely I've misunderstood how indices work in pandas; let me double check some pessimal cases and get back to you

Reply:
:+1: i think adding a test case for this logic based on those checks would be a good idea